File: ecommerc_dotnet/module/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ecommerc_dotnet.module;

public class Product
{
    public Guid id { get; set; }
    public string name { get; set; }
    
    [Column(TypeName = "TEXT")]
    public string description { get; set; }
    public string  thmbnail { get; set; }
    public Guid subcategoryId { get; set; }
    public Guid storeId { get; set; }
    public decimal price { get; set; }
    public int? quanity { get; set; } = null;
    
    [Column(TypeName = "Timestamp")]
    public DateTime createdAt { get; set; }

    [Column(TypeName = "Timestamp")] 
    public DateTime? updatedAt { get; set; } = null;
    
    public ICollection<ProductVarient> productVarients { get; set; }
    public ICollection<ProductImage> productImages { get; set; }
    
   public SubCategory subCategory { get; set; }
   public Store store { get; set; }
}

File: eccorce_app/app/src/main/java/com/example/eccomerce_app/model/Product.kt
package com.example.eccomerce_app.model

import java.util.UUID

data class Product(
    var id: UUID,
    var name: String,
    var description:String,
    var thmbnail:String,
    var subcategory_id: UUID,
    var store_id: UUID,
    var price: Double,
    var productVarients:List<List<ProductVarient>>?=null,
    var productImages:List<String>
) ;